src/api/utils/tradeQuoter.ts

```diff
@@ -0,0 +1,451 @@
+      calldata,
+    } = await this.fetchZeroExQuote( // fetchQuote (and switch...)
+      fromTokenAddress,
+      toTokenAddress,
+      fromTokenRequestAmount,
+      setOnChainDetails.manager,
+      (setOnChainDetails as any).totalSupply, // Typings incorrect,
+      chainId,
+      isFirmQuote,
+      slippagePercentage,
+      feeRecipient,
+      excludedSources
+    );
+
+    // Sanity check response from quote APIs
+    this.validateQuoteValues(
+      setOnChainDetails,
+      fromTokenAddress,
+      toTokenAddress,
+      fromUnits,
+      toUnits
+    );
+
+    const gas = await this.estimateGasCost(
+      options.tradeModule,
+      fromTokenAddress,
+      fromUnits,
+      toTokenAddress,
+      toUnits,
+      exchangeAdapterName,
+      fromAddress,
+      calldata,
+      setOnChainDetails.manager
+    );
+
+    const coinGecko = new CoinGeckoDataService(chainId);
+    const coinPrices = await coinGecko.fetchCoinPrices({
+      contractAddresses: [this.chainCurrencyAddress(chainId), fromTokenAddress, toTokenAddress],
+      vsCurrencies: [ USD_CURRENCY_CODE, USD_CURRENCY_CODE, USD_CURRENCY_CODE ],
+    });
+
+    if (!options.gasPrice) {
+      const gasOracle = new GasOracleService(chainId);
+      options.gasPrice = await gasOracle.fetchGasPrice();
+    }
+
+    return {
+      from: fromAddress,
+      fromTokenAddress,
+      toTokenAddress,
+      exchangeAdapterName,
+      calldata,
+      gas: gas.toString(),
+      gasPrice: options.gasPrice.toString(),
+      slippagePercentage: this.formatAsPercentage(slippagePercentage),
+      fromTokenAmount: fromUnits.toString(),
+      toTokenAmount: toUnits.toString(),
+      display: {
+        inputAmountRaw: options.rawAmount.toString(),
+        inputAmount: amount.toString(),
+        quoteAmount: fromTokenRequestAmount.toString(),
+        fromTokenDisplayAmount: this.tokenDisplayAmount(fromTokenAmount, options.fromTokenDecimals),
+        toTokenDisplayAmount: this.tokenDisplayAmount(toTokenAmount, options.toTokenDecimals),
+        fromTokenPriceUsd: this.tokenPriceUsd(
+          fromTokenAmount,
+          fromTokenAddress,
+          options.fromTokenDecimals,
+          coinPrices
+        ),
+        toTokenPriceUsd: this.tokenPriceUsd(
+          toTokenAmount,
+          toTokenAddress,
+          options.toTokenDecimals,
+          coinPrices
+        ),
+        gasCostsUsd: this.gasCostsUsd(options.gasPrice, gas, coinPrices, chainId),
+        gasCostsChainCurrency: this.gasCostsChainCurrency(options.gasPrice, gas, chainId),
+        feePercentage: this.formatAsPercentage(feePercentage),
+        slippage: this.calculateSlippage(
+          fromTokenAmount,
+          toTokenAmount,
+          fromTokenAddress,
+          toTokenAddress,
+          options.fromTokenDecimals,
+          options.toTokenDecimals,
+          coinPrices
+        ),
+      },
+    };
+  }
+
+  private sanitizeAddress(fromToken: Address, toToken: Address, fromAddress: Address) {
```

Comment:
This sanitization was only useful so that we could map to the token list for the decimals in a standardized way. if the decimals are already being passed in then this is not necessary